src/tiledb/cloud/asset.py

```diff
@@ -0,0 +1,40 @@
+"""An asset may be an array or a group."""
+
+from functools import partial
+from typing import Union
+
+import tiledb  # type: ignore
+
+from . import array  # type: ignore
+from . import groups  # type: ignore
+from .rest_api.models import ArrayInfo  # type: ignore
+from .rest_api.models import GroupInfo  # type: ignore
+
+
+def delete(uri: str, recursive: bool = False) -> None:
+    """Deregister the asset and remove its physical groups and arrays from storage.
+
+    :param uri: URI of the asset.
+    :return: None.
+    """
+    delete_map = {
+        "array": array.delete_array,
+        "group": partial(groups.delete, recursive=recursive),
+    }
+    asset_type = tiledb.object_type(uri, ctx=tiledb.cloud.Ctx())
+    func = delete_map[asset_type]
```

Comment:
Instead of, e.g.,

```python
if asset_type == "array":
    return array.delete_array(uri)
elif asset_type == "group":
    return groups.delete(uri)
else:
    raise Exception(...)
```

Note that moving these maps up to the module scope complicates unit testing. Maps inside the function allows us to patch public API tiledb.cloud functions as late as we want.

Reply:
I totally support your arguments here.
Another option here to avoid recreating the mappings in every function call would be to store them in a Class. 
Example just for further discussion: 
```python
class Asset:
    def __init__(self, ...):
        self.delete_map = {
           "array": array.delete_array,
           "group": partial(groups.delete, recursive=recursive),
        }
        self.info_map = {"array": array.info, "group": groups.info}

    def delete(self, uri: str, recursive: bool = False) -> None:
        asset_type = tiledb.object_type(uri, ctx=tiledb.cloud.Ctx())
        func = self.delete_map[asset_type]
        return func(uri)

    ....


asset = Asset()
```

Also mappings etc. can be modified through `__init__`